lighthouse-core/config/budgets.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * @license Copyright 2016 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+class Budgets {
+  /**
+ * @param {LH.Budgets.ResourceBudget} resourceBudget
+ * @return {LH.Budgets.ResourceBudget}
+ */
+  static validateResourceBudget(resourceBudget) {
+    const validResourceTypes = [
+      'total',
+      'document',
+      'script',
+      'stylesheet',
+      'image',
+      'media',
+      'font',
+      'other',
+      'thirdParty',
+    ];
+    if (!validResourceTypes.includes(resourceBudget.resourceType)) {
+      throw new Error(`Invalid resource type: ${resourceBudget.resourceType}. \n` +
+        `Valid resource types are: ${ validResourceTypes.join(', ') }`);
+    }
+    if (isNaN(resourceBudget.budget)) {
+      throw new Error('Invalid budget: ${resourceBudget.budget}');
+    }
+    return resourceBudget;
+  }
+
+  /**
+ * @param {LH.Budgets.TimingBudget} timingBudget
+ * @return {LH.Budgets.TimingBudget}
+ */
+  static validateTimingBudget(timingBudget) {
+    const validTimingMetrics = [
+      'firstContentfulPaint',
```

Comment:
since we're baking these into the format (which makes sense), we'll probably have to bikeshed the strings.

My first inclination is to have them match the audit IDs as those are already part of the public API: https://github.com/GoogleChrome/lighthouse/blob/f2bd3b6d473824a4e4c05a8a0844a0f3ecf119dc/lighthouse-core/config/default-config.js#L351-L356

But I can understand going camelcase here, too.